src/ssh/azext_ssh/tests/latest/test_custom.py

```diff
@@ -0,0 +1,196 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import io
+from knack import util
+import mock
+import unittest
+
+from azext_ssh import custom
+
+
+class SshCustomCommandTest(unittest.TestCase):
+    @mock.patch('azext_ssh.custom._do_ssh_op')
+    @mock.patch('azext_ssh.custom.ssh_utils')
+    def test_ssh_vm(self, mock_ssh_utils, mock_do_op):
+        cmd = mock.Mock()
+        custom.ssh_vm(cmd, "rg", "vm", "ip", "public", "private")
+
+        mock_do_op.assert_called_once_with(
+            cmd, "rg", "vm", "ip", "public", "private", mock_ssh_utils.start_ssh_connection)
+
+    @unittest.skip('have problems with patching functools.partial, will enable it after getting the root cause.')
+    @mock.patch('azext_ssh.custom._do_ssh_op')
+    @mock.patch('azext_ssh.custom.ssh_utils')
+    @mock.patch('functools.partial')
+    def test_ssh_config(self, mock_partial, mock_ssh_utils, mock_do_op):
```

Comment:
Hi @rlrossiter , which python version are you using to run this test ? It always failed from my side (python 3.8).

Reply:
The version isn't the problem, the code changed and broke the unit tests. A lot of them are now failing for me too because of the code changes (Python 3.6).

Reply:
Hi @rlrossiter, what other tests also failed for you ? As you can see from the CI log, all other tests are passed. Did you use the latest `azure-cli` dev branch and rerun `azdev setup` ? It should be only this test failed. And the reason has something to do with `functions.partial`. I'm not sure the root cause currently, will investigate deeper.

Reply:
I guess I didn't have all of the commits you added to this branch. Will re-pull and take a look if I get a chance.